dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/SignIn/Register/PreferredNamePage.cshtml.cs

```diff
@@ -45,7 +45,8 @@ public async Task<IActionResult> OnPost()
             return this.PageWithErrors();
         }
 
-        HttpContext.GetAuthenticationState().OnPreferredNameSet(PreferredName ?? ExistingName);
+        var preferredName = HasPreferredName == true ? PreferredName : ExistingName;
+        HttpContext.GetAuthenticationState().OnPreferredNameSet(preferredName!);
```

Comment:
Fixes an issue with not necessarily overwriting the preferred name with authenticationState.GetName() when returning to the page